Custom generators for plain ruby classes

redtachyons opened the pull request:
1. Service object
2. Form object
3. Query object

tachyons committed:
Added generators for

1. Service object
2. Form object
3. Query object

coderhs commented on lib/generators/form_object/USAGE:
```diff
@@ -0,0 +1,8 @@
+    rails generate form_object Thing
+
+    This will create:
+        app/services/thing.rb
```
If its a form object then we need it in `app/forms`

coderhs commented on lib/generators/rspec/form_object/form_object_generator.rb:
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+module Rspec
+class FormObjectGenerator < Rails::Generators::NamedBase
```
Intending.

coderhs commented on lib/generators/service_object/templates/service_object.rb.tt:
```diff
@@ -0,0 +1,18 @@
+<% module_namespacing do -%>
+class <%= class_name %>
```
In the main template and project, in general, I would like to keep format `ClassNameService`, didn't want to break the convention in clickx but here would like to keep the postfix service.

coderhs committed:
loofah vulnerability update

coderhs committed:
Rails 5.2 upgrade

coderhs committed:
Rename nomenclature use service, form & query generators

The proper name of these patters comes with the prefix , but at a
programmer convinience point of view using just terms like service, form and query
makes the CLI more convinient.

Also giveing the class name with the prefix Service, Query and Form. And using terms like ThingService, ThingQuery and ThingForm would improve the redability of our code. Thus I propose these changes in nomenclature.

coderhs commented:
@tachyons I have made changes to your `Custom Generator`. Basically, I changed the nomenclature used. Do share your review.

redtachyons commented:
LGTM
